src/main/resources/bpe/ping.bpmn

```diff
@@ -1,77 +1,123 @@
+        <bpmn:incoming>Flow_03hkxbe</bpmn:incoming>
+        <bpmn:outgoing>Flow_0wmpprs</bpmn:outgoing>
+      </bpmn:serviceTask>
+      <bpmn:exclusiveGateway id="Gateway_0z5gz96" default="Flow_1j54c2s">
+        <bpmn:incoming>Flow_1ipvu5v</bpmn:incoming>
+        <bpmn:outgoing>Flow_1j54c2s</bpmn:outgoing>
+        <bpmn:outgoing>Flow_101sqed</bpmn:outgoing>
+      </bpmn:exclusiveGateway>
+      <bpmn:sequenceFlow id="Flow_1j54c2s" sourceRef="Gateway_0z5gz96" targetRef="EventBasedGateway" />
+      <bpmn:serviceTask id="LogSendError" name="log send error" camunda:class="dev.dsf.bpe.service.LogSendError">
+        <bpmn:incoming>Flow_101sqed</bpmn:incoming>
+        <bpmn:outgoing>Flow_16ssf4a</bpmn:outgoing>
       </bpmn:serviceTask>
+      <bpmn:sequenceFlow id="Flow_101sqed" name="not-allowed or not-reachable" sourceRef="Gateway_0z5gz96" targetRef="LogSendError">
+        <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">${execution.hasVariable('statusCode') &amp;&amp; (statusCode == 'not-allowed' || statusCode == 'not-reachable')}</bpmn:conditionExpression>
+      </bpmn:sequenceFlow>
+      <bpmn:intermediateThrowEvent id="SendPing" name="ping">
+        <bpmn:extensionElements />
+        <bpmn:incoming>Flow_0v2ascf</bpmn:incoming>
+        <bpmn:outgoing>Flow_1ipvu5v</bpmn:outgoing>
+        <bpmn:messageEventDefinition id="MessageEventDefinition_03fdhw9" camunda:class="dev.dsf.bpe.message.SendPing">
+          <bpmn:extensionElements>
+            <camunda:field name="instantiatesCanonical">
+              <camunda:string>http://dsf.dev/bpe/Process/pong|#{version}</camunda:string>
+            </camunda:field>
+            <camunda:field name="profile">
+              <camunda:string>http://dsf.dev/fhir/StructureDefinition/task-ping|#{version}</camunda:string>
+            </camunda:field>
+            <camunda:field name="messageName">
+              <camunda:string>ping</camunda:string>
+            </camunda:field>
+          </bpmn:extensionElements>
+        </bpmn:messageEventDefinition>
+      </bpmn:intermediateThrowEvent>
+      <bpmn:sequenceFlow id="Flow_0v2ascf" sourceRef="SubProcessStartEvent" targetRef="SendPing" />
+      <bpmn:sequenceFlow id="Flow_1ipvu5v" sourceRef="SendPing" targetRef="Gateway_0z5gz96" />
+      <bpmn:exclusiveGateway id="Gateway_09z7b0z">
+        <bpmn:extensionElements />
+        <bpmn:incoming>Flow_0wmpprs</bpmn:incoming>
+        <bpmn:incoming>Flow_136htek</bpmn:incoming>
+        <bpmn:incoming>Flow_16ssf4a</bpmn:incoming>
+        <bpmn:outgoing>Flow_1ho1hys</bpmn:outgoing>
+      </bpmn:exclusiveGateway>
+      <bpmn:sequenceFlow id="Flow_0wmpprs" sourceRef="LogPong" targetRef="Gateway_09z7b0z" />
+      <bpmn:sequenceFlow id="Flow_136htek" sourceRef="LogNoResponse" targetRef="Gateway_09z7b0z" />
+      <bpmn:sequenceFlow id="Flow_16ssf4a" sourceRef="LogSendError" targetRef="Gateway_09z7b0z" />
+      <bpmn:endEvent id="SubProcessEndEvent" camunda:asyncBefore="true">
```

Comment:
Why is in the subprocess endEvent an `asyncBefore="true"` needed?

Reply:
Never mind, it is explained here: https://camunda.com/blog/2014/07/advanced-asynchronous-continuations/